docs/design.md

```diff
@@ -0,0 +1,38 @@
+# Why request-local-storage?
+
+Request local storage handles a particular problem, that of
+[high arity functions](https://en.wikipedia.org/wiki/Arity) in request/response
```

Comment:
It's both arity and also _depth_: How many levels of function calls does a parameter need to be passed through to reach where it's actually used?